<a href="https://colab.research.google.com/github/Ajit-bit-eng/Python_Practical_Assignments.ipynb/blob/main/STATISTICS_PRACTICAL.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**STATISTICS PRACTICAL **



8. Question: You have two sets of data representing the incomes of two different professions

Profession A: [48, 52, 55, 60, 62]

Profession B: [45, 50, 55, 52, 47]

Perform an F-test to determine if the variances of the two professions'
incomes are equal. What are your conclusions based on the F-test?

Task: Use Python to calculate the F-statistic and p-value for the given data.

Objective: Gain experience in performing F-tests and interpreting the results in terms of variance comparison.

ANS.

To perform an F-test to compare the variances of the two professions' incomes, we can follow these steps in Python:

Calculate the variances of both datasets.

Compute the F-statistic, which is the ratio of the larger variance to the smaller variance.

Calculate the p-value based on the F-distribution.

Let’s implement this in Python.

In [1]:
import numpy as np
from scipy.stats import f

# Given data
profession_a = [48, 52, 55, 60, 62]
profession_b = [45, 50, 55, 52, 47]

# Step 1: Calculate the sample variances
var_a = np.var(profession_a, ddof=1)  # variance of Profession A
var_b = np.var(profession_b, ddof=1)  # variance of Profession B

# Step 2: Calculate the F-statistic (larger variance / smaller variance)
if var_a > var_b:
    f_statistic = var_a / var_b
    dfn, dfd = len(profession_a) - 1, len(profession_b) - 1  # degrees of freedom for A and B
else:
    f_statistic = var_b / var_a
    dfn, dfd = len(profession_b) - 1, len(profession_a) - 1

# Step 3: Calculate the p-value
p_value = 2 * (1 - f.cdf(f_statistic, dfn, dfd))  # two-tailed p-value

# Display the results
print("F-statistic:", f_statistic)
print("p-value:", p_value)


F-statistic: 2.089171974522293
p-value: 0.49304859900533904


The F-statistic for comparing the variances of the two professions' incomes is approximately 2.09, and the p-value is approximately 0.493.

Interpretation:
Since the p-value (0.493) is much greater than a typical significance level (e.g., 0.05), we fail to reject the null hypothesis that the variances of the two professions' incomes are equal. This suggests that there is no statistically significant difference in the income variances of the two professions based on this test.

9. Question: Conduct a one-way ANOVA to test whether there are any statistically significant differences in
average heights between three different regions with the following data:

Region A: [160, 162, 165, 158, 164]
Region B: [172, 175, 170, 168, 174]
Region C: [180, 182, 179, 185, 183]

Task: Write Python code to perform the one-way ANOVA and interpret the results.
Objective: Learn how to perform one-way ANOVA using Python and interpret F-statistic and p-value

ANS.

In [2]:
from scipy.stats import f_oneway

# Given data for each region
region_a = [160, 162, 165, 158, 164]
region_b = [172, 175, 170, 168, 174]
region_c = [180, 182, 179, 185, 183]

# Perform one-way ANOVA
f_statistic, p_value = f_oneway(region_a, region_b, region_c)

f_statistic, p_value


(67.87330316742101, 2.870664187937026e-07)

Interpretation:

Since the p-value is much smaller than a typical significance level (e.g., 0.05), we reject the null hypothesis. This indicates that there are statistically significant differences in average heights between the three regions.